### PR TITLE
Postfix deref (rebased version of #11)

### PR DESCRIPTION
**Pull request by zmughal**
Alternative to <https://github.com/neilb/Perl-MinimumVersion/pull/11>.

This is rebased in order to be ready to merge.

### CODE REVIEW

**Comment by zmughal** [comment]
@trwyant Does this look right?

**Comment by trwyant** [comment]
Sorry for the delay -- Real Life has a way of intruding. It looks like your version is a refactor of Perl::MinimumVersion that strips out the code in _postfix_deref() that handled the PPI mis-parses that existed when I originiated my pull request. Unfortunately I don't have time to review your version adequately until this evening at the earliest. But I wanted you to know your message was received, and thank you for working with me on this, and on Perl::MinimumVersion in general.

**Comment by zmughal** [comment]
Oh, I hadn't meant to remove code!

There are two commits here:
1. the first one keeps all the checks in place without bumping the minimum PPI version; this is the same as the current tip #11. This can be verified with:
   ```shell
   vimdiff <( git show fad5755e6ec4500d4e100e745903788736a7fc27:lib/Perl/MinimumVersion.pm ) \
           <( git show 8282683bb2482a0a149e6ea041c2f18bee25c557:lib/Perl/MinimumVersion.pm )
   ```
2. the second one only keeps what was originally working with PPI-1.237_001 and bumps the PPI version to 1.238.

If the second one is still broken when under PPI-1.238, then could you give examples of where it fails? Is this something that would need to be patched in PPI?

**Comment by trwyant** [comment]
It has been 5-ish years since I wrote the original code. Without going back and completely re-engineering it to get my brain fully up to speed, what I can say is:

- I believe t/27_postfix_deref.t rings all the changes in the syntax -- at least I intended it to do so. And as you know, [4d0bac4](https://github.com/neilb/Perl-MinimumVersion/commit/4d0bac4bf61d800ccfe20a8ceb3cf56f10a2b80b) passes. I really think this is the most important evidence that the new code is good, but --

- I took a good look at your code, and it looks to me like all the pieces/parts are there. You turned my code back-to-front, which makes sense since we know PPI will find the deref.

My original mish-mash of anonymous subs and chained if-elsif reflects the struggle to extract the necessary data out of PPI's mis-parse. Thank you for doing the clean-up that was enabled by bumping the PPI version.

In short, I believe the branch is ready to be merged. Thanks.

**Comment by Grinnz** [comment]
Thank you!
